hnn_core/network.py

```diff
@@ -374,23 +380,31 @@ def add_evoked_drive(self, name, *, mu, sigma, numspikes,
         if not self._legacy_mode:
             _check_drive_parameter_values('evoked', sigma=sigma,
                                           numspikes=numspikes)
-
         drive = _NetworkDrive()
         drive['type'] = 'evoked'
         if name == 'extgauss':
             drive['type'] = 'gaussian'  # XXX needed to pass legacy tests!
-        drive['cell_specific'] = True
+        if n_drive_cells == 'n_cells':
+            n_drive_cells = self.n_cells
+            drive['cell_specific'] = True
+        else:
+            _validate_type(n_drive_cells, types=int)
+            _check_option('n_drive_cells', n_drive_cells, allowed_values=[1])
+            drive['cell_specific'] = False
+        drive['n_drive_cells'] = n_drive_cells
         drive['seedcore'] = seedcore
 
-        drive['dynamics'] = dict(mu=mu, sigma=sigma, numspikes=numspikes,
-                                 sync_within_trial=sync_within_trial)
+        drive['dynamics'] = dict(mu=mu, sigma=sigma, numspikes=numspikes)
         drive['events'] = list()
 
         self._attach_drive(name, drive, weights_ampa, weights_nmda, location,
-                           space_constant, synaptic_delays)
+                           space_constant, synaptic_delays,
+                           n_drive_cells=n_drive_cells,
+                           cell_specific=drive['cell_specific'])
 
     def add_poisson_drive(self, name, *, tstart=0, tstop=None, rate_constant,
-                          location, weights_ampa=None, weights_nmda=None,
+                          location, n_drive_cells='n_cells',
+                          weights_ampa=None, weights_nmda=None,
```

Comment:
Although it's not entirely necessary, I've added the `n_drive_cells` arg here to establish API consistency across the `add_xxx_drive()` methods and to create a method to implement noise correlations across cells in a spontaneously firing network. While the latter is not yet a mainstream use-case of hnn-core, I'm interested in exploring this further in my research personally.